File: sample-app/ContosoJobs/Services/ICompanyService.cs
using DataAccess.Entities;
using ContosoJobs.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ContosoJobs.Services
{
    public interface ICompanyService
    {
        // TODO uncomment these as needed
        Task<bool> RegisterNewCompany(CompanyInfo company);
        Task<string> GetCompanyByUserId(string userId);
        Task CreateCompanyUserRel(string userId, string companyId);
        Task<IEnumerable<CompanyUserInfo>> GetUsersByCompanyId(string companyId);
        Task<IEnumerable<JobPostingInfo>> GetJobPostingsByCompanyId(string companyId);
        Task<string> CreateUser(ApplicationUser user);
        Task<bool> AddUserToRole(string userId, string userRole);
        //Task<IEnumerable<CompanyRating>> GetAllCompanyRatingsAsync();
        //Task<IEnumerable<CompanyRating>> GeCompanyRatingByCompanyIdAsync(string companyId);
        Task<CompanyInfo> GetCompanyByCompanyId(string companyId);
        //Task<CompanyRating> GetDocumentAsync(string id, string companyId);
        //Task UpdateDocumentAsync(CompanyRating companyRating);
    }
}

File: sample-app-java/src/main/java/com/sample/contosojobs/ContosoJobsApplication/service/ICompanyService.java
package com.sample.contosojobs.ContosoJobsApplication.service;

import java.util.List;

import com.sample.contosojobs.ContosoJobsApplication.Entity.CompanyInfo;
import com.sample.contosojobs.ContosoJobsApplication.Entity.CompanyJobInfo;
import com.sample.contosojobs.ContosoJobsApplication.Entity.CompanyUserInfo;
import com.sample.contosojobs.ContosoJobsApplication.Entity.JobPostingInfo;
import com.sample.contosojobs.ContosoJobsApplication.Entity.RegisterCompanyViewModel;
import com.sample.contosojobs.ContosoJobsApplication.Entity.UserInfo;
import com.sample.contosojobs.ContosoJobsApplication.Entity.Users;

public interface ICompanyService {

	 public int CreateCompany(RegisterCompanyViewModel company) throws InterruptedException;
	 public void CreateCompanyUser(CompanyUserInfo company);
	 public String GetCompanyIdByUser(String id);
	 public List<CompanyJobInfo> GetJobPostingsByCompanyId(String companyId);
	public List<Users> GetUsersByCompanyId(String companyId);
	public CompanyInfo getCompanyByCompanyId(String companyId);
}
